src/main/scala/com/joeybaruch/datamodel/AggregatedMetrics.scala

```diff
@@ -0,0 +1,133 @@
+package com.joeybaruch.datamodel
+
+import scala.math.{max, min}
+
+object AggregatedMetrics {
+
+  //  trait AggregatableMetric{
+  //    val eventCount: Long
+  //    val earliestTimestamp: Long
+  //    val latestTimestamp: Long
+  //
+  //    lazy val timeSpan = latestTimestamp - earliestTimestamp
+  //
+  //    def +(that: AggregatableMetric): AggregatableMetric
+  //  }
+
+  // implicit class lets us define our own methods on existing classes - like Map[String, Long]
+  implicit class NamedCountersCollection(val map: Map[String, Long]) {
+    def unionWith(that: NamedCountersCollection): Map[String, Long] = {
+      val kvPairsConcat: Seq[(String, Long)] = this.map.toList ++ that.map.toList
+      val groupedByKey = kvPairsConcat.groupBy(_._1)
+      val mergedByValue = groupedByKey.map { case (key, kvSeq) => key -> kvSeq.foldLeft(0L)(_ + _._2) }
+      mergedByValue
+    }
+  }
+
+  implicit class NamedCountersHypercollection(val map: Map[String, Map[String, Long]]) {
+    def unionWith(that: NamedCountersHypercollection): Map[String, Map[String, Long]] = {
+      val kvPairsConcat: Seq[(String, Map[String, Long])] = this.map.toList ++ that.map.toList
+      val groupedByKey = kvPairsConcat.groupBy(_._1)
+      val mergedByValue = groupedByKey.map { case (key, strAndNamedCountersSeq) =>
+        key -> strAndNamedCountersSeq.foldLeft(Map.empty[String, Long])(_ unionWith _._2)
+      }
+      mergedByValue
+    }
+  }
```

Comment:
These are nice ways to merge two maps, but I would also like it to function as
`type NamedCountersHypercollection = Map[String, Long]` 
this way when I call other map methods on the class fields below, they'll work just fine.

Any Ideas, or should I just settle for the full name below?

Reply:
Figured this out - leave it for now